core-rust/state-manager/src/transaction/ledger_transaction.rs

```diff
@@ -130,9 +130,33 @@ const GENESIS_TRANSACTION_SYSTEM_TRANSACTION_DISCRIMINATOR: u8 = 1;
 
 #[derive(Debug, Clone, PartialEq, Eq, ManifestCategorize, ManifestEncode, ManifestDecode)]
 pub struct FlashTransactionV1 {
+    pub nonce: u64,
     pub state_updates: StateUpdates,
 }
 
+impl FlashTransactionV1 {
+    pub fn prepare(&self) -> Result<PreparedFlashTransactionV1, PrepareError> {
+        let state_updates_hash = hash(scrypto_encode(&self.state_updates).unwrap());
+        let flash_hash = HashAccumulator::new()
+            .update([
+                TRANSACTION_HASHABLE_PAYLOAD_PREFIX,
+                TransactionDiscriminator::V1Flash as u8,
+            ])
+            .update(state_updates_hash)
+            .update(self.nonce.to_le_bytes().as_slice())
+            .finalize();
+
+        Ok(PreparedFlashTransactionV1 {
+            state_updates: self.state_updates.clone(),
+            summary: Summary {
+                effective_length: 0,
```

Comment:
These two fields aren't correct... Probably doesn't matter a ton as I'm not sure they're used, BUT imo it's better to get rid of this method, move the content to `prepare_as_full_body_child`, and use:

```
let ((name, state_updates), summary) = ConcatenatedDigest::prepare_from_transaction_child_struct::<(
                SummarizedRawFullBody<String>,
                SummarizedRawFullBody<StateUpdates>,
            )>(decoder, TransactionDiscriminator::V1Flash)?;
```

It's more efficient, consistent, and the summary is more accurate.

See e.g. the implementation for SystemTransaction:
```
impl TransactionFullChildPreparable for PreparedSystemTransactionV1 {
    fn prepare_as_full_body_child(decoder: &mut TransactionDecoder) -> Result<Self, PrepareError> {
        let ((prepared_instructions, blobs, pre_allocated_addresses, hash_for_execution), summary) =
            ConcatenatedDigest::prepare_from_transaction_child_struct::<(
                PreparedInstructionsV1,
                PreparedBlobsV1,
                PreparedPreAllocatedAddresses,
                PreparedHash,
            )>(decoder, TransactionDiscriminator::V1System)?;
        Ok(Self {
            encoded_instructions: manifest_encode(&prepared_instructions.inner.0)?,
            references: prepared_instructions.references,
            blobs,
            pre_allocated_addresses,
            hash_for_execution,
            summary,
        })
    }
}
```

Reply:
:+1: done in https://github.com/radixdlt/babylon-node/pull/803